app/qml/Highlight.qml

```diff
@@ -152,7 +152,13 @@ Item {
         elements.push( componentMoveTo.createObject( objOwner, { "x": data[ dataStartIndex ], "y": data[ dataStartIndex + 1 ] } ) )
         for ( let i = dataStartIndex + 2; i < data.length; i += 2 )
         {
-          elements.push( componentLineTo.createObject( objOwner, { "x": data[ i ], "y": data[ i + 1 ] } ) )
+          if (data[ i ] === geometryType) {
```

Comment:
instead of checking if data[i] == geometryType (which is potentially fragile), we should modify the for-cycle...

For each ring we have the number of points in it (the second entry after geometry type). So let's use that count to have something like this:
```
while ( i < data.length )
{
  geomType = data[i++];
  count = data[i++];
  // ... add moveTo ...
  for ( k = 1; k < count; ++k )
  {
    // ... add lineTo ...
  }
}
```